File: AdminUi/test/AdminUi.Tests.Integration/Models/UpdateClientResponse.cs
﻿namespace Backbone.AdminUi.Tests.Integration.Models;
public class UpdateClientResponse
{
    public required string ClientId { get; set; }
    public required string DisplayName { get; set; }
    public required string DefaultTier { get; set; }
    public required DateTime CreatedAt { get; set; }
    public int? MaxIdentities { get; set; }
}

File: Applications/AdminUi/packages/admin_api_types/lib/src/clients/update_client_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'update_client_response.g.dart';

@JsonSerializable()
class UpdateClientResponse {
  final String clientId;
  final String displayName;
  final String defaultTier;
  final DateTime createdAt;
  final int? maxIdentities;

  UpdateClientResponse({
    required this.clientId,
    required this.displayName,
    required this.defaultTier,
    required this.createdAt,
    required this.maxIdentities,
  });

  factory UpdateClientResponse.fromJson(dynamic json) => _$UpdateClientResponseFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$UpdateClientResponseToJson(this);
}
